src/cycax/cycad/cycad_part.py

```diff
@@ -163,18 +163,18 @@ def make_counterdrill(self, side: str, diameter: float, depth: float, angle: flo
         msg = "The adding of counterdrill to a side has not been implemented."
         raise NotImplementedError(msg)
 
-    def test_mesh_hole(self, x: float, y: float, z: float, diamter: float, depth: float):
+    def test_mesh_hole(self, x: float, y: float, z: float, diameter: float, depth: float):
         if (
-            x - diamter / 2 < self.x_min
-            or x + diamter / 2 > self.x_max
-            or y - diamter / 2 < self.y_min
-            or y + diamter / 2 > self.y_max
+            x - diameter / 2 < self.x_min
```

Comment:
consider using some memory and pre-calculate radius and use that in the equations.